packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,227 @@
+  } else if (type === 'telephone') {
+    keyboardAndAutocompleteProps.keyboardType = 'telephone';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'done';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType =
+      autoCompleteSuggestionType ?? 'telephone';
+  } else if (type === 'email') {
+    keyboardAndAutocompleteProps.keyboardType = 'email';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'done';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'email';
+  } else if (type === 'url') {
+    keyboardAndAutocompleteProps.keyboardType = 'url';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'go';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  } else if (type === 'numeric') {
+    keyboardAndAutocompleteProps.type = 'text';
+    /* the default keyboardType:numeric shows alphanumeric keyboard on iOS but number pad on android. making it type:text and keyboardType:decimal fixes this on all platforms.
+     * source: https://css-tricks.com/everything-you-ever-wanted-to-know-about-keyboardType/#aa-decimal
+     */
+    keyboardAndAutocompleteProps.keyboardType = 'decimal';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'done';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  } else if (type === 'search') {
+    keyboardAndAutocompleteProps.keyboardType = 'search';
+    keyboardAndAutocompleteProps.keyboardReturnKeyType = keyboardReturnKeyType ?? 'search';
+    keyboardAndAutocompleteProps.autoCompleteSuggestionType = autoCompleteSuggestionType ?? 'none';
+  }
+
+  return keyboardAndAutocompleteProps;
+};
+
+// need to do this to tell TS to infer type as TextInput of React Native and make it believe that `ref.current.clear()` exists
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isReactNative = (_textInputRef: any): _textInputRef is TextInputReactNative => {
+  return getPlatformType() === 'react-native';
+};
+
+export const TextInput = ({
+  label,
+  labelPosition = 'top',
+  placeholder,
+  type = 'text',
+  defaultValue,
+  name,
+  value,
+  maxCharacters,
+  onChange,
+  onBlur,
+  isDisabled,
+  necessityIndicator,
+  validationState,
+  errorText,
+  helpText,
+  successText,
+  isRequired,
+  icon,
+  prefix,
+  showClearButton,
+  onClearButtonClick,
+  isLoading,
+  suffix,
+  autoFocus,
+  keyboardReturnKeyType,
+  autoCompleteSuggestionType,
+}: TextInputProps): ReactElement => {
+  const textInputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
+  const [shouldShowClearButton, setShouldShowClearButton] = useState(false);
+
+  const renderInteractionElement = (): ReactNode => {
+    if (isLoading) {
+      return <InfoIcon size="medium" color="surface.text.subtle.lowContrast" />;
```

Comment:
Add a TODO comment here. Need to change with spinner.

Reply:
fixed in 717f7a1da44ddbaf76edc37e72d60e182d95576b